### ISSUE TITLE
Traefik with rancher-metadata backend is not behaving like with rancher API backend

### ISSUE THREAD

**Issue by fabiorauber** [containous/traefik#1852]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Using traefik with the latest experimental build (containous/traefik:experimental) to test rancher-metadata integration. 

### What did you expect to see?

Each frontend having its own backend, like when using integration via Rancher API, even though every backend container is named 'plone'. 
Traefik 1.3 configured with Rancher API creates several backends based on the stack name, like "backend-site1-plone", "backend-site2-plone", etc.

### What did you see instead?

Each frontend is pointing to a single backend, "backend-plone", when using integration via rancher-metadata.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
containous/traefik:779eeba650b1bd9a9228cf17da7ed4ba02ed93c9 
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yml
traefik:
  ports:
    - 8000:8080/tcp
    - 80:80/tcp
    - 443:443/tcp
  labels:
    io.rancher.scheduler.global: 'true'
    io.rancher.scheduler.affinity:host_label: role=traefik
    io.rancher.scheduler.affinity:container_label_ne: io.rancher.stack_service.name=$${stack_name}/$${service_name}
  image: containous/traefik:779eeba650b1bd9a9228cf17da7ed4ba02ed93c9
  command:
    - "--web"
    - "--entryPoints=Name:http\ Address::80\ Redirect.EntryPoint:https"
    - "--entryPoints=Name:https\ Address::443\ TLS"
    - "--defaultentrypoints=http,https"
    - "--rancher"
    - "--rancher.exposedbydefault=false"
    - "--rancher.metadata=true"
    - "--acme"
    - "--acme.email=email@domain.com"
    - "--acme.storage=/mnt/acme.json"
    - "--acme.entrypoint=https"
    - "--acme.ondemand=true"
    - "--logLevel=INFO"
  volumes:
    - "traefik_acme:/mnt:nocopy"

```
Every traefik backend container has the following rancher labels:

    labels:
      traefik.port: 8080
      traefik.enable: true
      traefik.frontend.rule: "Host:www.domain.com,domain.com"
      traefik.acme.domains: "www.domain.com,domain.com"

**Comment by fabiorauber** [comment]
Just tested with latest experimental build (765c44d77f0e94336cd8d61dfeda59a08f77504c / raclette), and the issue persists.

**Comment by Guilhem30** [comment]
I have the same issue on 1.4 rc1. A single backend for every containers with the same name.
Rancher should take into account the stack-name too

**Comment by fabiorauber** [comment]
Version 1.4-rc2 also tested, and same behavior persists. It does not consider stack name in backend name generation.

**Comment by SantoDE** [comment]
Ill have a Look at that later on :)

**Comment by fabiorauber** [comment]
Thank you very much @SantoDE! It is preventing me from using Traefik with rancher-metadata instead of API, making much harder to take rancher down for maintenance. =D

**Comment by SantoDE** [comment]
Hey @fabiorauber,

just to confirm - are u using the same rancher version as in #2100 ?

**Comment by fabiorauber** [comment]
Exactly @SantoDE. Rancher 1.4.3

**Comment by SantoDE** [comment]
Once you upgraded, could you test again? (looking into that anyway as @Guilhem30 reported the same behavior)

**Comment by SantoDE** [comment]
I can confirm this bug. I've opened a pull request to merge a fix. 

I would appreciate if you could do a pre test with an experimental test docker image @fabiorauber 
santode/traefik:issue_1852

:)

**Comment by fabiorauber** [comment]
Will do first thing tomorrow @SantoDE! Thanks!

**Comment by Guilhem30** [comment]
After some tests, your experimental image seem to fix the bug. Traefik works perfectly for me with rancher metadata and your image. Thanks you for your patch @SantoDE ! Well done!

**Comment by fabiorauber** [comment]
I confirm what @Guilhem30 stated. The backends are being named with the stack name prefix. Now I can schedule that Rancher upgrade with my sites online =D

**Comment by traefiker** [comment]
Closed by #2107.
